Source/WebCore/inspector/InspectorCanvas.cpp

```diff
@@ -892,6 +893,9 @@ Ref<Protocol::Canvas::Canvas> InspectorCanvas::buildObjectForCanvas(bool capture
         if (is<WebGL2RenderingContext>(m_context))
             return Protocol::Canvas::ContextType::WebGL2;
 #endif
+        if (is<GPUCanvasContext>(m_context))
+            return Protocol::Canvas::ContextType::WebGPU;
```

Comment:
why was this included as part of this PR?  i ask because this is not how WebGPU inspection was done in the past (see [215342@main](https://commits.webkit.org/215342@main)), so it's really odd for it to be done this way now

Reply:
It was included because opening web inspector resulted in a crash otherwise when WebGPU was being used in a web worker

Reply:
i see.  my guess is that this is because inside `InspectorCanvasAgent::internalEnable` we're iterating over all `CanvasRenderingContext::instances()` and attempting to create an `InspectorCanvas` for each of them rather than only the ones that Web Inspector currently supports (see callers of `InspectorInstrumentation::didCreateCanvasRenderingContext`)

id suggest undoing this part of the PR and instead add either
```diff
-#if ENABLE(OFFSCREEN_CANVAS)
-            if (is<PlaceholderRenderingContext>(context))
-                continue;
-#endif
+             if (!is<CanvasRenderingContext2D(context))
+#if ENABLE(OFFSCREEN_CANVAS)
+                 && !is<OffscreenCanvasRenderingContext>(context)
+#endif
+                 && !is<WebGLRenderingContextBase>(context)
+                 && !is<ImageBitmapRenderingContext(context))
+                 continue;
```
or
```
if (is<GPUCanvasContext>(context))
    continue;
```
because when the time comes to actually add back support for WebGPU in Web Inspector, it'll likely not be done using the `GPUCanvasContext` for the same reasons outlined in [215342@main](https://commits.webkit.org/215342@main)

Reply:
Repro can be observed by removing these web inspector changes, opening https://webgpu.github.io/webgpu-samples/samples/worker, and then opening web inspector

Happy to consider other alternatives if there is a better option 👍

Reply:
Oh great thank you @dcrousso! I’ll give that a try

Reply:
https://github.com/WebKit/WebKit/pull/19064